packages/@aws-cdk/aws-glue/lib/job-executable.ts

```diff
@@ -0,0 +1,415 @@
+   */
+  readonly extraFiles?: string[];
+
+  /**
+   * Setting this value to true prioritizes the customer's extra JAR files in the classpath.
+   *
+   * @default - extra jars are not prioritized.
+   *
+   * @see `--user-jars-first` in https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly extraJarsFirst?: boolean;
+}
+
+/**
+ * Props for creating a Scala Spark (ETL or Streaming) job executable
+ */
+export interface ScalaJobExecutableProps extends SharedSparkJobExecutableProps {
+  /**
+   * The fully qualified Scala class name that serves as the entry point for the job.
+   *
+   * @see `--class` in https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly className: string;
+}
+
+/**
+ * Props for creating a Python Spark (ETL or Streaming) job executable
+ */
+export interface PythonJobExecutableProps extends SharedSparkJobExecutableProps {
+
+  /**
+   * The Python version to use.
+   */
+  readonly pythonVersion: PythonVersion;
+
+  /**
+   * The Amazon S3 paths to additional Python modules that AWS Glue adds to the Python path before executing your script.
+   * Only individual files are supported, not a directory path.
+   *
+   * @default - no extra python files and argument is not set
+   *
+   * @see `--extra-py-files` in https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly extraPythonFiles?: string[];
+
+  /**
+   * Setting this value to true prioritizes the customer's extra JAR files in the classpath.
+   *
+   * @default - priortiy is not given to extra jars and argument is not set
+   * @see `--user-jars-first` in https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly extraJarsFirst?: boolean;
+}
+
+/**
+ * Props for creating a Python shell job executable
+ */
+export interface PythonShellExecutableProps extends SharedJobExecutableProps {
+
+  /**
+   * The Python version to use.
+   */
+  readonly pythonVersion: PythonVersion;
+
+  /**
+   * The Amazon S3 paths to additional Python modules that AWS Glue adds to the Python path before executing your script.
+   * Only individual files are supported, not a directory path.
+   *
+   * @default - no extra python files and argument is not set
+   *
+   * @see `--extra-py-files` in https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly extraPythonFiles?: string[];
+}
+
+/**
+ * The executable properties related to the Glue job's GlueVersion, JobType and code
+ *
+ * TODO test for exceptions
+ */
+export class JobExecutable {
+
+  /**
+   * Create Scala executable props for Apache Spark ETL job.
+   *
+   * @param props Scala Apache Spark Job props
+   */
+  public static etlScala(props: ScalaJobExecutableProps): JobExecutable {
```

Comment:
minor: I'd like the names of these static methods to be reversed, as I think it sounds better. this would be "scalaEtl"